shipyard_blueprints/hex/shipyard_hex/cli/hex_run_project.py

```diff
@@ -48,145 +22,35 @@ def get_args():
-        print("Could not connect to Hex with the provided API token and Project Id.")
-        return ec.EXIT_CODE_AUTHENTICATION_ERROR
-
-
-def determine_run_status(run_response):
-    """Determine the status of the run and return the appropriate exit code."""
-    status = run_response["status"]
-    end_time = run_response["endTime"]
-    run_id = run_response["runId"]
-
-    status_messages = {
-        "COMPLETED": ec.EXIT_CODE_COMPLETED,
-        "KILLED": ec.EXIT_CODE_KILLED,
-        "PENDING": ec.EXIT_CODE_PENDING,
-        "RUNNING": ec.EXIT_CODE_RUNNING,
-        "UNABLE_TO_ALLOCATE_KERNEL": ec.EXIT_CODE_UNABLE_TO_ALLOCATE_KERNEL,
-        "ERRORED": ec.EXIT_CODE_ERRORED,
-    }
-
-    print(f"Hex reports that run {run_id} has a status of {status}.", end=" ")
-    if end_time:
-        print(f"End time: {end_time}")
-    else:
-        print()
-    return status_messages.get(status, ec.EXIT_CODE_UNKNOWN_ERROR)
-
-
 def main():
     args = get_args()
     project_id = args.project_id.strip()
     api_token = args.api_token.strip()
+    wait = shipyard.args.convert_to_boolean(args.wait_for_completion)
+    client = HexClient(api_token=api_token, project_id=project_id)
 
     ## run the project
-    trigger_run = run_project(project_id, api_token)
-    status_code = trigger_run.status_code
-    response_json = trigger_run.response_json
-
-    ## create artifacts folder to save runId on success and reason if not successful
-    base_folder_name = shipyard.logs.determine_base_artifact_folder("hex")
-    artifact_subfolder_paths = shipyard.logs.determine_artifact_subfolders(
-        base_folder_name
-    )
-    shipyard.logs.create_artifacts_folders(artifact_subfolder_paths)
-
-    if status_code == 201:
-        run_id = response_json[
-            "runId"
-        ]  ## need this to verify the status in the other blueprint
-        shipyard.logs.create_pickle_file(
-            artifact_subfolder_paths, "runId", run_id
-        )  ## save the run id
-        ## save the response
-        response_data_file = shipyard.files.combine_folder_and_file_name(
-            artifact_subfolder_paths["responses"],
-            f"project-{project_id}_run_{run_id}_response.json",
-        )  ## save the run response
-        shipyard.files.write_json_to_file(response_json, response_data_file)
-    else:
-        reason = response_json["reason"]
-        shipyard.logs.create_pickle_file(artifact_subfolder_paths, "reason", reason)
-    if args.wait_for_completion.upper() == "TRUE":
-        response = get_run_status(project_id, api_token, run_id)
-        status_exit_code = determine_run_status(response)
-
-        while status_exit_code in (ec.EXIT_CODE_RUNNING, ec.EXIT_CODE_PENDING):
-            print(f"Run {run_id} is still running. Checking again in 60 seconds")
+    run_response = client.run_project()
+    if run_response.ok:
+        logger.info("Successfully triggered hex project")
+    run_id = run_response.json()["runId"]
```

Comment:
should have the client return the json response
this should be something along the lines of
```
run_details = client.run_project(project_id)
run_id = run_detail.get("runId")
```